quota/etcd/quotapb/quotapb.proto

```diff
@@ -133,6 +133,10 @@ message UpdateConfigRequest {
   // Fields modified by the update request.
   // For example: "state" or "max_tokens".
   google.protobuf.FieldMask update_mask = 3;
+
+  // If true the updated quota is reset, regardless of the updates's contents.
+  // A reset quota is replenished to its maximum number of tokens.
+  bool reset_quota = 4;
```

Comment:
**Re "single monster DoSomething request":** that's a strong exaggeration. We're pretty far from it. In fact, this may be one of the most style-compliant APIs we have in the project.

Do recall that we're modelling a REST API, so it's not absurd that a `PATCH quotas/my/quota/config` could alter quota state. All in all, I don't think this is abuse of the method/paradigm. If you feel too strongly about it, we could go with a custom `PATCH quotas/my/quota/config:reset`, but I'm not sold that's worth it, nor I think this is the PR to add it.

The "odd special case" is a single condition that's clearly documented, at least from what I can tell. I think it brings enough benefit to merit being there. "Dangling" fields to control certain update behaviors are relatively common in APIs; here's an example from [App Engine's UpdateService method](https://cloud.google.com/appengine/docs/admin-api/reference/rest/v1/apps.services/patch).

**Re "using the same name to refer to different resources":** Resetting a quota by its name seems reasonable to me; what other name would you use?

**Re exposing token count through config:** that's an interesting idea. It plays well in a resource-oriented world. I'll put something together and see how it looks like. (That's also an argument for separate storage and API protos, btw.)

**Re documentation:** I've no excuses here. I'll add details and push a new commit.

I didn't expect this to come as such a surprise. We reviewed this design and I described both functionality and API, down to proto representations. Being able to reset is an oncall-support functionality first and foremost, and the current API allows it by a simple `curl http://my-server/quotas/my/quota/config?reset_quota=true`, which seems perfectly fine in theory (and practice!) to me.